[trivial] Add missing const qualifiers.

domob1812 opened the pull request:
Add some `const` qualifiers to references that are not modified and should be marked as const.

MarcoFalke commented on src/txmempool.h:
```diff
@@ -672,10 +672,10 @@ class CTxMemPool
 
 public:
-    CCoinsViewMemPool(CCoinsView *baseIn, CTxMemPool &mempoolIn);
+    CCoinsViewMemPool(CCoinsView *baseIn, const CTxMemPool &mempoolIn);
```
nit: If you do trivial fixes, please also do all other trivial fixes in the same line:

```diff
diff --git a/src/main.cpp b/src/main.cpp
index ed34137..2777245 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1471 +1471 @@ bool GetTransaction(const uint256 &hash, CTransaction &txOut, const Consensus::P
-            const CCoinsViewCache &view = *pcoinsTip;
+            const CCoinsViewCache& view = *pcoinsTip;
diff --git a/src/txmempool.h b/src/txmempool.h
index c991514..d17e332 100644
--- a/src/txmempool.h
+++ b/src/txmempool.h
@@ -675 +675 @@ protected:
-    const CTxMemPool &mempool;
+    const CTxMemPool& mempool;
@@ -678 +678 @@ public:
-    CCoinsViewMemPool(CCoinsView *baseIn, const CTxMemPool &mempoolIn);
+    CCoinsViewMemPool(CCoinsView* baseIn, const CTxMemPool& mempoolIn);

domob1812 replied:
I agree but was not aware that there is a general style preference of `uint256& hash` over `uint256 &hash`.  I personally like the former better, but did not want to reformat "someone else's" code.  Is the former explicitly preferred in Bitcoin sources?  (Which are not always consistent in style.)

MarcoFalke replied:
It is not really important to get every white space "correct" but you can read about the general preferred code style over here: https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#developer-notes

domob1812 replied:
Thanks for the heads-up, I've updated the patch now.  This is not explicitly mentioned in the doc, but the example code block uses the `uint256& hash` style.

MarcoFalke replied:
If you don't want to do that manually, you can just do `$ git diff HEAD~ -U0|cfd`. Where `cfd` is an alias for  `./contrib/devtools/clang-format-diff.py -p1 -i -v`.

domob1812 committed:
[trivial] Add missing const qualifiers.

Add some const qualifiers to references that are not modified and should
be marked as const.

dcousens commented:
utACK f7c4f78

paveljanik commented:
ACK https://github.com/bitcoin/bitcoin/commit/f7c4f79f07c777801db7dc047cd45eaf2bba81c9

MarcoFalke commented:
utACK f7c4f79
